models/common.py

```diff
@@ -18,16 +18,20 @@
 import requests
 import torch
 import torch.nn as nn
+from IPython.display import display
```

Comment:
Shouldn't ipython be an optional dependency ? IMO, it would make sense to remove it for a minimal setup without "displaying" results.

Reply:
@vfdev-5 ipython is only used for checking if environment is notebook, clearing notebook cells and showing images in notebook cells. We have in requirements due to the popularity of your Colab notebook. 

Does it conflict with anything or is it installing significant other dependencies in your environment?

Reply:
Thanks for the response, @glenn-jocher 

> Does it conflict with anything or is it installing significant other dependencies in your environment?

well, in terms of space, ipython + deps does not take much space (~1-2MB), I agree. I start to understand better all the context of this work. For demos in colab etc it totally makes sense. In terms of headless server usage (probably less frequent?), it feels weird to install ipython, matplotlib, seaborn (it is just my opinion).

Reply:
@vfdev-5 got it. Much of this is the difference between training support, visualization support, export support etc. We try to strike the right balance but it's impossible to please everyone.